testing/run_tests.py

```diff
@@ -239,6 +294,13 @@ def main():
     dart_filter = args.dart_filter.split(',') if args.dart_filter else None
     RunDartTests(build_dir, dart_filter)
 
+  if 'java' in types and not IsWindows():
```

Comment:
nit: We may want to throw or assert that it's not windows, rather than just skipping silently.

Reply:
I think that makes sense, and @matthew-carroll mentioned this too. But none of the other cases do this so I didn't want this type to be the one case that decided to assert and fail on Windows if you picked `all`. @chinmaygarde do you have an opinion on this? I could change them all at once here.

Reply:
No opinion. I think my python-fu is rubbish. Please feel free to edit as necessary.

Reply:
Done.